onnxruntime/core/optimizer/selectors_actions/selector_action_transformer.cc

```diff
@@ -32,33 +33,34 @@ void SelectorActionRegistry::RegisterSelectorAndAction(const std::string& name,
   const Entry& entry = name_to_entry_it->second;
   for (const auto& [op_type, versions] : entry.ops_and_versions) {
     ORT_UNUSED_PARAMETER(versions);
-    op_type_to_entry_.emplace(op_type, &entry);
+    op_type_to_entry_.emplace(op_type + domain, &entry);
```

Comment:
if we go with this approach, would be good to use something like `pair<string, string>` or at least separate them with a delimiter. if the op type and domain can be arbitrary strings the concatenation of one pair could be the same as another distinct pair.

Reply:
I thought about picking pair<string, string> as the key. However I thought it might add the complexity of the hash table.

Reply:
In this case we are not uniquely identifying the op, (Domain, op, version) in the Key. The key is (domain, operator) the version is in Entry which maps different versions.

Reply:
I assumed we wouldn't use a map of op to version if we switched to using OpIdentifier. 

Given the primary usage is in MatchAndProcess by this code, if we use OpIdentifier we can just do a direct lookup instead of lookup of op followed by matching of version. 

```
      // check the supported versions if specified
      const auto& versions = entry->ops_and_versions.find(node.OpType())->second;
      if (!versions.empty()) {
        if (std::find(versions.cbegin(), versions.cend(), node.SinceVersion()) == versions.cend()) {
          continue;
        }
      }
```

OpIdentifier may need the ability to do a wildcarded match of the version but that could probably be implemented in an operator== method to keep things simple.

Reply:
I was not planning on changing the existing functionality. My aim is to make this work for domains other than the default domain, kOnnxDomain.

Reply:
If we want to minimize changes, the selector could apply any filtering on domain instead of pushing that up to the transformer level. i.e. the filtering on op and opset in MatchAndProcess can be seen as a simplistic check, with the implementation of the selector doing additional things like domain/EP validation. With that a lot of the changes in the PR may not be necessary. 

This might also be better given potentially the EP involved needs to be checked as well before deciding in the fusion can proceed. 

e.g. the CUDA EP how has some NHWC ops, and the ConvActivationFusion EP list in GenerateTransformers includes that EP. do you know if the CUDA EP expects a Conv in the internal NHWC domain to be fused?